src/lifeblood/net_messages/message_processor.py

```diff
@@ -215,23 +224,36 @@ async def new_message_received(self, message: Message) -> bool:
          each group should have single tcp connection, otherwise no guarantee about ordering
         """
         destination = message.message_destination().split_address()
-        if destination[0] != self.__address:
+        if destination[0] not in self.__addresses:
             self._logger.error('received message not meant for me, dropping')
             return True
 
-        if len(destination) > 1:  # redirect it further
-            dcurrent, dnext = destination[0], destination[1:]
-            assert dcurrent == self.__address
+        si = 0
+        for si, addr in enumerate(destination):
+            if addr not in self.__addresses:
+                break
+        else:
+            si += 1
+
+        current_part = destination[:si]
+        assert len(current_part) > 0  # destination check above catches this error, so assert must never fail
+        next_part = destination[si:]
+        if len(next_part) > 0:  # redirect it further
+            current_part = current_part[:1]  # throw out any other our addresses, including dups
```

Comment:
check if current_part is already properly normalized, give debug message is address was renormalized